### ISSUE TITLE
toUIImage returns nil for iOS

### ISSUE THREAD

**Issue by Danijal98** [icerockdev/moko-resources#640]
I am using the version: `0.23.0`
I have the following structure for my images:  `shared/commonMain/resources/MR/images/ic_partnership.svg`

I have a suggested helper method
```
fun getImageByFileName(name: String): ImageResource {
    val fallbackImage = MR.images.transparent
    return MR.images.getImageByFileName(name) ?: fallbackImage
}
```

From iOS I call it like this
```
let imageRes = ResourcesKt.getImageByFileName(name: "ic_partnership") // Successfully finds the resource
let uiImage = imageRes.toUIImage() // returns nil
Image(uiImage: uiImage!).padding() // crashes because "Unexpectedly found nil while unwrapping an Optional value"
```

**Comment by Alex009** [comment]
svg works in samples. svg on iOS require iOS 13+, check your iOS version please

**Comment by Danijal98** [comment]
> svg works in samples. svg on iOS require iOS 13+, check your iOS version please

I am using the iOS 17, so I think that should not be the issue

**Comment by Alex009** [comment]
@Danijal98 please test your svg inside xcode project directly. maybe iOS for some reason can't process this svg

**Comment by djehrlich** [comment]
I’m seeing a similar issue with an SVG file on iOS.  Interestingly, while in light mode, I see the image (.toUIImage() returns non-nil).  However, when I switch to dark mode, I don’t see the image (.toUIImage() returns nil).  Maybe this clue helps?

**Comment by rph8** [comment]
> I’m seeing a similar issue with an SVG file on iOS. Interestingly, while in light mode, I see the image (.toUIImage() returns non-nil). However, when I switch to dark mode, I don’t see the image (.toUIImage() returns nil). Maybe this clue helps?

Is it maybe related to this one (should be fixed in the most recent version / beta)?
https://github.com/icerockdev/moko-resources/issues/543

**Comment by djehrlich** [comment]
> > I’m seeing a similar issue with an SVG file on iOS. Interestingly, while in light mode, I see the image (.toUIImage() returns non-nil). However, when I switch to dark mode, I don’t see the image (.toUIImage() returns nil). Maybe this clue helps?
> 
> Is it maybe related to this one (should be fixed in the most recent version / beta)? #543

Dark mode SVG images are indeed working.  The comments for [#624](https://github.com/icerockdev/moko-resources/pull/624) provided the insight I needed - add an image resource with “-dark” appended to its name.  I tested by copying  “car_black.svg” to “car_black-dark.svg” and both light and dark mode now work as advertised.

**Comment by Danijal98** [comment]
@Alex009 I can confirm that this bug does not occur anymore for me after upgrading to moko 0.24.1
